training/tests/test_sign_in_up.py

```diff
@@ -1,76 +1,85 @@
+            self.assertEqual(client.post('/register', data=dict(username='Fernando4', email='Fernando4@g.com', password='Fernando', confirm='Fernando', accept_tos=True), follow_redirects=True).status_code, 201)
 
             # We checked if all users were created and saved in the database
             for i in range(1, 5):
                 username = "Fernando" + str(i)
                 self.assertNotEqual(None, User.query.filter_by(id=username).first())
 
-    def test_sing_up_some_users_and_bad_credentials(self):
-        print("-------------------------------")
-        print("ESTE TEST HAY QUE DESAROLLARLO")
-        print("-------------------------------")
-        assert True
-        # ERROR DESCUBIERTO, No se verifica que el mail no exista en la abse al momento de hacer sign up
+    def test_sing_up_two_users_same_email(self):
         with app.test_client() as client:
             # We create two users, with same mail
-            #self.assertEqual(client.post('/index', data=dict(username='Fernando1', email='Fernando1@g.com', password='Fernando', confirm='Fernando', accept_tos=True), follow_redirects=True).status_code, 201)
-            #self.assertEqual(client.post('/index', data=dict(username='Fernando2', email='Fernando1@g.com', password='Fernando', confirm='Fernando', accept_tos=True), follow_redirects=True).status_code, 201)
-            pass
+            self.assertEqual(client.post('/register', data=dict(username='Fernando1', email='Fernando1@g.com', password='Fernando', confirm='Fernando', accept_tos=True), follow_redirects=True).status_code, 201)
+            self.assertEqual(client.post('/register', data=dict(username='Fernando2', email='Fernando1@g.com', password='Fernando', confirm='Fernando', accept_tos=True), follow_redirects=True).status_code, 420)
+
+    def test_sing_up_two_users_same_username(self):
+        with app.test_client() as client:
+            # We create two users, with same username
+            self.assertEqual(client.post('/register', data=dict(username='Fernando', email='Fernando1@g.com', password='Fernando', confirm='Fernando', accept_tos=True), follow_redirects=True).status_code, 201)
+            self.assertEqual(client.post('/register', data=dict(username='Fernando', email='Fernando2@g.com', password='Fernando', confirm='Fernando', accept_tos=True), follow_redirects=True).status_code, 420)
 
     # Login Tests
     def test_sign_in_one_user_check_session(self):
         with app.test_client() as client:
             # We create a User
-            self.assertEqual(client.post('/index', data=dict(username='Fernando', email='Fernando@g.com', password='Fernando', confirm='Fernando', accept_tos=True), follow_redirects=True).status_code, 201)
+            self.assertEqual(client.post('/register', data=dict(username='Fernando', email='Fernando@g.com', password='Fernando', confirm='Fernando', accept_tos=True), follow_redirects=True).status_code, 201)
```

Comment:
You can create a user directly in the database and assume it's there, for example in the setUp method. Also, if you're going to use the same block of code in many places please put it in a function.

Reply:
I have already created a function that creates a user and validates his email. I probably forgot to change that when I was doing those tests. I will improve it.